draft-ietf-cats-usecases-requirements.md

```diff
@@ -213,23 +202,73 @@ informative:
 
    This document makes use of the following terms:
 
-   Service:
-   :  An offering provided by a service provider, similar to the
-     notion of a 'service function' in {{?RFC7665}}, which may or may not
-     be of composite nature but appears in the problem space of CATS as
-     a single service to which traffic needs to be steered.
+  Client:
+     An endpoint that is connected to a service provider network.
 
-   Service instance:
-   :  A run-time environment (e.g., a server or a
-     process on a server) that makes a service available.  A particular
-     service could be made available at multiple service instances at
-     the same or different locations.
+   Computing-Aware Traffic Steering (CATS):  A traffic engineering
+     approach [I-D.ietf-teas-rfc3272bis] that takes into account the
+     dynamic nature of computing resources and network state to optimize
+     service-specific traffic forwarding towards a given service contact
+     instance.  Various relevant metrics may be used to enforce such
+     computing-aware traffic steering policies.
+
+   Service:  An offering that is made available by a provider by
+     orchestrating a set of resources (networking, compute, storage,
+     etc.).  Which and how these resources are solicited is part of the
+     service logic which is internal to the provider.  For example,
+     these resources may be:
+ 
+        * Exposed by one or multiple processes (a.k.a.  Service
+        Functions (SFs) ).  [RFC7665]
+ 
+        * Provided by virtual instances, physical, or a combination
+        thereof.
+
+        * Hosted within the same or distinct nodes.
+
+        * Hosted within the same or multiple service sites.
+
+        * Chained to provide a service using a variety of means.
+
+        How a service is structured is out of the scope of CATS.
+
+        The same service can be provided in many locations; each of them
+        constitutes a service instance.
```

Comment:
```suggestion
        constitutes a service instance.
        
Service contact instance:
:  A client-facing service function instance
      that is responsible for receiving requests in the context of a
      given service.  A service request is processed according to the
      service logic (e.g., handle locally or solicit backend resources).
      Steering beyond the service contact instance is hidden to both
      clients and CATS components.

 : A service contact instance is reachable via at least one Egress
      CATS Forwarder.

: A service can be accessed via multiple service contact instances
      running at the same or different locations (service sites).

: The same service contact instance may dispatch service requests to
      one or more service instances (e.g., an instance that behaves as a
      service load-balancer).
```

Reply:
If both service instance and service contact instance exist, from my understanding, service instance is not a very useful concept. In the document, there are a number of occurences of "selection of service or service selection" here and there. After introducing service contact instance, which is selected, service instance or service contact instance? More like service contact instance to me.

Reply:
> If both service instance and service contact instance exist, from my understanding, service instance is not a very useful concept. In the document, there are a number of occurences of "selection of service or service selection" here and there. After introducing service contact instance, which is selected, service instance or service contact instance? More like service contact instance to me.

Agree. service instance should be used in very few occurrences; what matters is the service contact instance.

Reply:
> If both service instance and service contact instance exist, from my understanding, service instance is not a very useful concept. In the document, there are a number of occurences of "selection of service or service selection" here and there. After introducing service contact instance, which is selected, service instance or service contact instance? More like service contact instance to me.

Agree. service instance should be used in very few occurrences; what matters is the service contact instance.